### DISCUSSION TITLE
MapParams endpoint url

### DISCUSSION THREAD

**Discussion by gauraw10** [goadesign/goa#2966] [Q&A] [answered]
What should be HTTP endpoint URL for the MapParams defined in [documentation](https://goa.design/v1/reference/goa/http/dsl/#MapParams.)

I tried like
```
Payload(func() {
			Attribute("test", MapOf(String, String), "The request test params.")
			Required("test")
		})
```
and

```
HTTP(func() {
			GET("/")
			MapParams("test")
			Response(StatusOK)
			Response("NotFound", StatusNotFound)
		})

```
I am using query parameter like:

`/?test={"email":"test@test.com", "test":"test"}`

I am not able to get this payload(map) in my code. Am I missing something?

**Comment by gauraw10** [comment, marked as answer]
I got the answer and it would be like the below
/?test[email]=test@test.com&test[test]=test

Thanks

**Reply by raphael** [reply]
Just to add to this: `MapParams` maps each query string parameter to a key in the corresponding map attribute. So in this example there is a single key `test` which itself is a map. So two things to note:
1. This use case does not require `MapParams`, it could simply be `Params("test")`. This is a normal case of encoding a map into a query string URL.
2. If the query string was changed to `?email=test@test.com&test=test` then `MapParams` would do what you expected in the first place (that is populate the map attribute which each query string parameter).

**Reply by gauraw10** [reply]
Thanks for your reply, Yes, my use case is the second option and I am able to query now.
